persistence/src/main/scala/za/co/absa/spline/persistence/model/entities.scala

```diff
@@ -84,14 +84,22 @@ object DBVersion {
   */
 case class DataSource(
   uri: DataSource.Uri,
+  name: DataSource.Name,
   override val _key: DataSource.Key
 ) extends Vertex with RootEntity {
-  def this() = this(null, null)
+  def this() = this(null, null, null)
 }
 
 object DataSource {
   type Key = ArangoDocument.Key
   type Uri = String
+  type Name = String
+
+  private val NameRegexp = "(?:.*/)?([^/]+)/*".r
```

Comment:
Why are the regexes here and in the script different?

Reply:
I just thought it was clearer to explicitly specify a non-capturing group to make the whole string match, not just a suffix. Essentially it makes no difference in the result but IMO more readable. But I can use the same one if you prefer.

Reply:
But if it's more readable, I would use it in the script as well.

By the way, for me, more readable is the one in the script. It's easier to see it will take the part at the end, here I have to consider the whole string.

Either way, I think it's better to use the same one on both places.